internal/storage/clickhouse.go

```diff
@@ -531,13 +532,78 @@ func getBlockNumbersStringArray(blockNumbers []*big.Int) string {
 }
 
 func (c *ClickHouseConnector) InsertBlockData(data []common.BlockData) error {
-	return nil
+	query := `INSERT INTO ` + c.cfg.Database + `.block_data (chain_id, block_number, data)`
+	batch, err := c.conn.PrepareBatch(context.Background(), query)
+	if err != nil {
+		return err
+	}
+	for _, blockData := range data {
+		blockDataJSON, err := json.Marshal(blockData)
+		if err != nil {
+			return err
+		}
+		err = batch.Append(
+			blockData.Block.ChainId,
+			blockData.Block.Number,
+			blockDataJSON,
+		)
+		if err != nil {
+			return err
+		}
+	}
+	return batch.Send()
 }
 
-func (c *ClickHouseConnector) GetBlockData(blockNumbers []*big.Int) (data []common.BlockData, err error) {
-	return nil, nil
+func (c *ClickHouseConnector) GetBlockData(qf QueryFilter) (blockDataList []common.BlockData, err error) {
+	query := fmt.Sprintf("SELECT data FROM %s.block_data FINAL WHERE block_number IN (%s) AND is_deleted = 0%s",
```

Comment:
shouldn't this include the chain_id as well?

Reply:
There are actually many places where chainID is missing from queries. Let's solve this in another PR to not clutter this one